protocol/configurable-calldata-gas.md

```diff
@@ -0,0 +1,77 @@
+# Configurable Calldata Gas Costs
+
+#### Metadata
+
+Authors: @niran
+Created: 2025-06-30
+
+---
+
+# Purpose
+ 
+This document proposes making calldata costs configurable by the chain operator at runtime, not just during hard forks.
+
+---
+
+# Summary
+
+[EIP 7623](https://eips.ethereum.org/EIPS/eip-7623) uses `STANDARD_TOKEN_COST` and `TOTAL_COST_FLOOR_PER_TOKEN` parameters to calculate the floor cost of each transaction. This proposal makes these parameters configurable via `SystemConfig` instead of hard-coded in the client.
+
+We introduce **two new configuration values** that can be modified via the rollup's `SystemConfig` contract:
+
+| Name | Type | Default | Meaning |
+|------|------|-------------------|---------|
+| `eip7623StandardTokenCost` | `uint32` | `4` | The ratio of the cost of a non-zero byte in `tx.data` to the cost of a zero byte in `tx.data` |
+| `eip7623TotalCostFloorPerToken` | `uint32` | `10` | The cost of a zero byte in `tx.data` when calculating the total cost floor |
```

Comment:
Can we use a uint16 or other smaller type? uint32 definitely seems like overkill for the first parameter, and we can almost certainly pack both of these into 32 or fewer bytes

Reply:
2f36340876b4371a9b283098ac2659e0f56994fc